### PR TITLE
Simplify method in DAppSigner

### PR DESCRIPTION
**Pull request by franfernandez20**
Enhance DAppSigner by adding methods that streamline communication for dapps, requiring just a Transaction or basic parameters and eliminating the need for developers to construct complex objects.

**Description**:
Add methods to facilitate the integration for dApps developers.  
 * support Transaction at signTransaction, signAndExecuteTransaction and executeTransaction methods
 * support only a string to sign messages
 * support only a string to sign and execute a query

**Related issue(s)**:

Fixes #

**Notes for reviewer**:
This pull request is part of PR #76, which will be permanently deleted once all topics are split.

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/lib/dapp/DAppSigner.ts [hunk 1]
```diff
@@ -89,15 +107,35 @@ export class DAppSigner implements Signer {
...
 
-  async populateTransaction<T extends Transaction>(transaction: T): Promise<T> {
+  async checkTransaction<T extends Transaction>(transaction: T): Promise<T> {
     throw new Error('Method not implemented.')
   }
 
-  async signTransaction<T extends Transaction>(transaction: T): Promise<T> {
+  async populateTransaction<T extends Transaction>(transaction: T): Promise<T> {
     throw new Error('Method not implemented.')
   }
```
[the remark this answers is not in the document]
Comment (franfernandez20) [reply src/lib/dapp/DAppSigner.ts]: I adapt to the last changes. This is outdated

**Review by zanctor — APPROVED** [review]

**Comment by teacoat** [comment]
Not sure I understand the intent of this - does it make sense to be putting stuff like this in the signer or should the be somewhere else? The signer has a pretty specific function and I'm not sure polluting it with unrelated functions makes sense.

File: src/examples/typescript/dapp/main.ts [hunk 2]
```diff
@@ -289,6 +289,66 @@ async function hedera_signTransaction(_: Event) {
...
+
+  const bodyBytes = Buffer.from(getState('signer-execute-transaction-body'), 'base64')
+  const sigMap = base64StringToSignatureMap(
+    getState('signer-execute-transaction-signature-map'),
+  )
+
+  const bytes = proto.Transaction.encode({ bodyBytes, sigMap }).finish()
+  const transaction = Transaction.fromBytes(bytes)
+
+  const signer = dAppConnector.getSigner(AccountId.fromString(accountId))
```
Comment (teacoat) [inline src/examples/typescript/dapp/main.ts:308]: For example here, would it make more sense to do something like:

```
return dAppConnector.executeTransaction(transaction, AccountId.fromString(accountId))
```
The signer should only be used for the intended purpose and not a dumping ground for general walletconnect helper functions

**Comment by teacoat** [comment]
Also need to update docs with this

File: src/examples/typescript/dapp/main.ts (hunk 2 above)
Comment (hgraphql) [reply src/examples/typescript/dapp/main.ts:308]: I think we could overload the functions in the dApp connector to provide this functionality there and not add additional methods to the signer implementation

**Comment by hgraphql** [comment]
> Not sure I understand the intent of this - does it make sense to be putting stuff like this in the signer or should the be somewhere else? The signer has a pretty specific function and I'm not sure polluting it with unrelated functions makes sense.

I think the intention is to be able to pass a more familiar object to these functions - Transaction, or a string for a message, or a query by name vs currently having to pass more esoteric params such as https://github.com/hashgraph/hedera-wallet-connect/blob/6df0ed00f8aa78cf8b687a49bccc67642c7d1ad0/src/lib/shared/payloads.ts#L54

File: src/examples/typescript/dapp/main.ts (hunk 2 above)
Comment (teacoat) [reply src/examples/typescript/dapp/main.ts:308]: @hgraphql I think that sounds like a good idea, keeping the signer strictly related to the hedera signer implementation would be wise imo

Comment (franfernandez20) [reply src/examples/typescript/dapp/main.ts:308]: > For example here, would it make more sense to do something like:
> 
> ```
> return dAppConnector.executeTransaction(transaction, AccountId.fromString(accountId))
> ```
> 
> The signer should only be used for the intended purpose and not a dumping ground for general walletconnect helper functions

This feature was initially part of the last pull request; however, we opted to divide it into separate contributions for better clarity and management. I concur that incorporating this function through dappConnector is a viable option.

Moreover, integrating it within the signer facilitates the management of multiple signers. Developers have the flexibility to initialize and manage it independently from the dappConnector if they choose to do so.

Comment (teacoat) [reply src/examples/typescript/dapp/main.ts:308]: @franfernandez20 If they want to use it independently from the dapp connector then they should be using the included functions which cover all of these things already - signWithSigner, executeWithSigner, etc.

This seems to just be adding redundant functionality to the signer? It seems like most of this stuff is already handled inside the dappconnector so I'm not sure what exactly this is improving

Comment (franfernandez20) [reply src/examples/typescript/dapp/main.ts:308]: Using executeWithSigner, you are not invoking the defined method `hedera_executeTransaction`. I understand your perspective, and perhaps `hedera_signAndExecuteTransaction` could be used through the call method. The idea was to allow users to create a transaction easily and use it consistently, regardless of the specific method.

**Comment by itsbrandondev** [comment]
hey everyone!

i'm providing an update on this PR:
* @franfernandez20 will be moving the functions to `dAppConnector` in alignment with feedback shared by @teacoat and others
* these changes will be made in the coming days, lets be ready to review

thanks : )

**Comment by franfernandez20** [comment]
> hey everyone!
> 
> i'm providing an update on this PR:
> 
> * @franfernandez20 will be moving the functions to `dAppConnector` in alignment with feedback shared by @teacoat and others
> * these changes will be made in the coming days, lets be ready to review
> 
> thanks : )



> hey everyone!
> 
> i'm providing an update on this PR:
> 
> * @franfernandez20 will be moving the functions to `dAppConnector` in alignment with feedback shared by @teacoat and others
> * these changes will be made in the coming days, lets be ready to review
> 
> thanks : )



After re-evaluating both objects, I believe it doesn't make sense to move these functions to the dAppConnector without also including them in the dAppSigner. The goal of the PR was to make it easier for developers to use these functions without needing to manage the network, account, or complex input parameters. However, moving these functions won't achieve this goal.

Therefore, my suggestion is to keep the PR as it is and ask for more arguments on why enhancing the Signer interface could be a problem. The dAppSigner object is designed to handle different sessions independently, which supports this goal.

If we can't reach a consensus on this, it makes more sense to me to reject the PR than to move these methods.

**Comment by hgraphql** [comment]
closing for now as we have not reached consensus, we'll continue to work to improve the dev experience on the repo

**Pull request closed without merging**